### ISSUE TITLE
Rechunk parameter not respected using read_parquet

### ISSUE THREAD

**Issue by jfinkle1016** [pola-rs/polars#8932]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

When reading in a parquet the `rechunk` parameter is not respected.

### Reproducible example

```python
import numpy as np
import polars as pl
from pyarrow.parquet import read_metadata

a = np.random.random((200, 5))
# We expect there to be fifty chunks
pl.DataFrame(a).write_parquet('tmp.pq', row_group_size=4)

# Should output `num_row_groups: 50`
print(read_metadata('tmp.pq'))

df = pl.read_parquet('tmp.pq', rechunk=True)

# We expect 1 but we get 50
print(df.n_chunks())

# We expect 1 and get 1
print(df.rechunk().n_chunks())
```

### Expected behavior

```python
import numpy as np
import polars as pl
from pyarrow.parquet import read_metadata

a = np.random.random((200, 5))
# We expect there to be fifty chunks
pl.DataFrame(a).write_parquet('tmp.pq', row_group_size=4)

# Should output `num_row_groups: 50`
print(read_metadata('tmp.pq'))

df = pl.read_parquet('tmp.pq', rechunk=True)

# We expect 1 but we get 50
print(df.n_chunks())

# We expect 1 and get 1
print(df.rechunk().n_chunks())
```

### Installed versions

<details>

```
--------Version info---------
Polars:      0.17.14
Index type:  UInt32
Platform:    Linux-5.15.0-1031-aws-x86_64-with-glibc2.31
Python:      3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:39:03) 
[GCC 11.3.0]

----Optional dependencies----
numpy:       1.23.4
pandas:      1.5.1
pyarrow:     10.0.1
connectorx:  <not installed>
deltalake:   <not installed>
fsspec:      2022.11.0
matplotlib:  3.7.1
xlsx2csv:    <not installed>
xlsxwriter:  <not installed>
```

</details>

**Comment by jfinkle1016** [comment]
@myz540 for visibility

**Comment by ritchie46** [comment]
fixed by #8935

**Comment by jfinkle1016** [comment]
Amazing. Thanks for the fast turn around
